lib/kafo_parsers/puppet_strings_module_parser.rb

```diff
@@ -0,0 +1,125 @@
+# encoding: UTF-8
+require 'json'
+require 'kafo_parsers/doc_parser'
+
+module KafoParsers
+  class PuppetStringsModuleParser
+    # You can call this method to get all supported information from a given manifest
+    #
+    # @param [ String ] manifest file path to parse
+    # @return [ Hash ] hash containing values, validations, documentation, types, groups and conditions
+    def self.parse(file)
+      content = new(file)
+      docs    = content.docs
+
+      # data_type must be called before other validations
+      data = {
+        :object_type => content.data_type,
+        :values      => content.values,
+        :validations => content.validations
+      }
+      data[:parameters] = data[:values].keys
+      data.merge!(docs)
+      data
+    end
+
+    def self.available?
+      `#{puppet_bin} help strings`
+      if $?.success?
+        return true
+      else
+        raise LoadError, "#{puppet_bin} does not have strings module installed"
+      end
+    rescue LoadError => e
+      raise KafoParsers::ParserNotAvailable.new(e)
+    end
+
+    def initialize(file)
+      @file = file
+      raise KafoParsers::ModuleName, "File not found #{file}, check your answer file" unless File.exists?(file)
+
+      command = "#{self.class.puppet_bin} strings #{file} --emit-json-stdout"
+      @raw_json = `#{command}`
+      unless $?.success?
+        raise KafoParsers::ParseError, "'#{command}' returned error\n#{@raw_json}"
+      end
+
+      begin
+        @complete_hash = ::JSON.parse(@raw_json)
+      rescue ::JSON::ParserError => e
+        raise KafoParsers::ParseError, "'#{command}' returned invalid json output: #{e.mesasge}"
+      end
+      self.data_type # we need to determine data_type before any further parsing
+
+      self
+    end
+
+    # AIO and system default puppet bins are tested for existence, fallback to `puppet` found in $PATH otherwise
+    def self.puppet_bin
+      @puppet_bin ||= begin
+        paths = [ '/opt/puppetlabs/bin', '/usr/bin/puppet' ]
```

Comment:
I took a different approach in https://github.com/theforeman/kafo/pull/139, preferring $PATH and using /opt/puppetlabs/bin as the last entry.  Which should we do?